tsfc/fem.py

```diff
@@ -233,6 +234,11 @@ def flat_index(ordered_deriv):
         return result[0]
 
 
+def zero_tracking_literal(table, ufl_element):
+    track_zeros = isinstance(ufl_element, (ufl.EnrichedElement, ufl.MixedElement))
```

Comment:
Shurely that is a choice that COFFEE should be making based on the nonzero bounds it is fed.  Rather than trying to pre-guess which tables would require zero tracking or not.